src/SIM/SIMoutput.C

```diff
@@ -674,7 +674,18 @@ bool SIMoutput::writeGlvS2 (const Vector& psol, int iStep, int& nBlock,
     opt.project.find(SIMoptions::GLOBAL) != opt.project.end();
 
   size_t sMAX = nf;
-  if (haveAsol) sMAX += nf;
+  size_t sAna = 0;
+  if (haveAsol) { // dummy evaluation to find number of components
+    Vec4 Xt;
+    Vector lovec;
+    if (mySol->hasScalarSol() == 3 || mySol->hasVectorSol() == 3)
+      myProblem->evalSol(lovec,*mySol->getStressSol(),Xt);
+    else if (this->getNoFields() == 1)
+      myProblem->evalSol(lovec,*mySol->getScalarSecSol(),Xt);
+    else
+      myProblem->evalSol(lovec,*mySol->getVectorSecSol(),Xt);
+    sMAX += sAna = lovec.size();
+  }
```

Comment:
Looks a bit overkill to do all this just to compute sMAX, which only needs to be an estimate (big enough). I guess in your case you have more than nf analytical solution components such that the old way failed (sID too small)? In that case I suggest to allow SID to grow beyond sMAX instead, i.e. replace l705-709 by something like:
```
  for (j = 1; j <= field.rows(); j++)
    if (!myVtf->writeNres(field.getRow(j),++nBlock,geomID))
      return false;
    else if (j <= sID.size())
      sID[j].push_back(nBlock);
    else
      sID.push_back(IntVec(1,nBlock));
 ```